### DISCUSSION TITLE
[UI] Knob-Control

### DISCUSSION THREAD

**Discussion by TopperDEL** [StereoKit/StereoKit#117] [Ideas]
For my app I would need a "Knob-Control" that changes a value by turning it left or right.

Do you think this would be feasable?

**Comment by maluoi** [comment]
Oh definitely, this should even be prototype-able from the C# side with the public APIs! I have a lot of UI things I'd like to try related to this, like light-switches and such. I just need to find a good chunk of time to do it! :) StereoKit's a little lean on UI elements right now, but I tried to provide a couple of tools for building atop what's there. The UI class has a couple methods that aren't fully documented that can make things a little easier!

One thing that does make me a little bit hesitant about a knob is that the MRTK doesn't have one of these! The team there has done a lot of really great interaction research while designing the MRTK, and so often, an omission like that can mean that there's something tricky hidden in the implementation or user experience. I do often like to re-validate these ideas myself though, as a different implementation, or even improvements in hand tracking quality can make a big difference!

Also, an HSlider often should functionally work as a substitute for a knob. Is that one missing something for you, or am I maybe thinking of a different type of knob-control?

**Reply by TopperDEL** [reply]
I was thinking the same - finger-tracking might not be accurate enough to support a knob like I imagine it. So maybe a Hslider migth be better. But it needs to be an "indefinit" slider as a knob could be turned indefinitely. But this could be achieved with a slider, that jumps back to zero if the touch ends. So you could use it to scratch back and forth.

Imagine one wants to change a day-field by turning a knob. You can turn and turn and with every tick the date changes to the next day (or the previous one if turned otherwise) and that never ends. I think a slider would work better with finger tracking. By sliding it to the right the days "tick through" one after another. If I release the touch, the value stops where it currently is. If the touch is active and I slide further to the right, the values tick through more quickly. That's something I would like to achieve.
A knob for it would be cool - but as said above, I also think that does not work very well.

I'll see if I can come up with something on my own. If it works good I may provide a PR to add that control to StereoKit.

**Reply by TopperDEL** [reply]
I basically switche to a "button left, button-right, value-in-the-middle"-approach. That works quite well, so the Knob-Control is not necessary anymore.

**Reply by maluoi** [reply]
Aha, an infinite slider is an excellent idea! For dates, maybe even one of those rolling cylinder lists... Some good ideas, yeah!

**Reply by TopperDEL** [reply]
In the long term a date-selection-control would be nice. But as I know have a completely own-coded UI, I would have to write this on my own anyway, I guess. :)
